### PR TITLE
ICU on alpine / macOS behaves differently

### PR DESCRIPTION
**Pull request by 6543**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by 6543** [comment]
address https://gitlab.alpinelinux.org/6543/aports/-/jobs/982040

by up-streaming patch included in https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/44647

**Comment by clokep** [comment]
This matches what I saw on macOS too: https://github.com/matrix-org/synapse/issues/15124#issuecomment-1438901931

**Comment by 6543** [comment]
@clokep should I adjust the commit to this new title or yours and forcepush (after tests did run)?

**Comment by 6543** [comment]
... and the linter looks like to have a `.` for the changelog ...

**Comment by clokep** [comment]
> @clokep should I adjust the commit to this new title or yours and forcepush (after tests did run)?

We squash merge so the commit names don't really matter.

**Comment by 6543** [comment]
🤔 does it close #15124 ?

or should we keep it open for now ...

**Comment by clokep** [comment]
> 🤔 does it close #15124 ?
> 
> or should we keep it open for now ...

I think we should keep it open, but would be curious what @reivilibre thinks.

**Comment by 6543** [comment]
tests passed 🚀

File: tests/storage/test_user_directory.py [hunk 1]
```diff
@@ -696,6 +696,8 @@ def test_icu_word_boundary_punctuation(self) -> None:
                 ["lazy'fox", "jumped", "over", "the", "dog"],
                 # ICU 70 on Ubuntu 22.04
                 ["lazy'fox", "jumped:over", "the.dog"],
+                # ICU 2.10.2 on Alpine edge / macOS
```
Comment (DMRobertson) [inline tests/storage/test_user_directory.py:699]: This looks like a pyicu version, not a libicu version like the other two above.

**Comment by reivilibre** [comment]
> > thinking does it close #15124 ?
> > or should we keep it open for now ...
> 
> I think we should keep it open, but would be curious what @reivilibre thinks.

Yeah, keep it open. I'm not extremely thrilled about having a library that doesn't behave in a predictable way for the one task we rely on it for :P.

File: tests/storage/test_user_directory.py (hunk 1 above)
Comment (6543) [reply tests/storage/test_user_directory.py]: @DMRobertson done

**Review by clokep — APPROVED** [review]

**Comment by clokep** [comment]
Thanks!

**Pull request merged**